scripts-available/CDB_FederatedServerDiagnostics.sql

```diff
@@ -0,0 +1,243 @@
+--------------------------------------------------------------------------------
+-- Private functions
+--------------------------------------------------------------------------------
+
+--
+-- Import a foreign table if it does not exist
+--
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_Import_If_Not_Exists(server_internal name, remote_schema name, remote_table name)
+RETURNS void
+AS $$
+DECLARE
+    local_schema name := @extschema@.__CDB_FS_Create_Schema(server_internal, remote_schema);
+BEGIN
+    IF NOT EXISTS (
+        SELECT * FROM pg_class
+        WHERE relnamespace = (SELECT oid FROM pg_namespace WHERE nspname = local_schema)
+        AND relname = remote_table
+    ) THEN
+        EXECUTE format('IMPORT FOREIGN SCHEMA %I LIMIT TO (%I) FROM SERVER %I INTO %I',
+                    remote_schema, remote_table, server_internal, local_schema);
+    END IF;
+END
+$$
+LANGUAGE PLPGSQL VOLATILE PARALLEL UNSAFE;
+
+--
+-- Get the version of a remote PG server
+--
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_Foreign_Server_Version_PG(server_internal name)
+RETURNS text
+AS $$
+DECLARE
+    remote_schema name := 'pg_catalog';
+    remote_table name := 'pg_settings';
+    local_schema name := @extschema@.__CDB_FS_Create_Schema(server_internal, remote_schema);
+    remote_server_version text;
+BEGIN
+    PERFORM @extschema@.__CDB_FS_Import_If_Not_Exists(server_internal, remote_schema, remote_table);
+
+    BEGIN
+        EXECUTE format('
+            SELECT setting FROM %I.%I WHERE name = ''server_version'';
+        ', local_schema, remote_table) INTO remote_server_version;
+    EXCEPTION WHEN OTHERS THEN
```

Comment:
may this be `WHEN insufficient_privilege`?

Reply:
I'd leave it "as is" for now, cause there may be other errors types (on the fdw category)